Ss body parser docs (#2438)
nestjs/docs.nestjs.com: content/middlewares.md

> info **Hint** The `configure()` method can be made asynchronous using `async/await` (e.g., you can `await` completion of an asynchronous operation inside the `configure()` method body).

> > warning **Warning** When using the `express` adapter, the NestJS app will register `json` and `urlencoded` from the package `body-parser` by default. This means if you want to customize that middleware per route, you need to turn off the global middleware by setting the `bodyParser` flag to `false` when creating the application with `NestFactory.create()`.

Reviewer: ```suggestion
> warning **Warning** When using the `express` adapter, the NestJS app will register `json` and `urlencoded` from the package `body-parser` by default. This means if you want to customize that middleware via the `MiddlewareConsumer`, you need to turn off the global middleware by setting the `bodyParser` flag to `false` when creating the application with `NestFactory.create()`.
```

You can still do per route middleware via `app.use()` in the same syntax as you would with express. `app.use('/some-path', middleware)`. Being explicit that this is only for the `MiddlewareConsumer` I think is a better option here

await app.listen(3000);
```

> #### Default body-praser middleware
> The nest application registers `json` and `urlencoded` from the package `body-parser` by default. This can be turned off by setting the `bodyParser` flag when creating the application. The options used when registering the middleware can be seen in the `registerParserMiddleware` function in the (express)[https://github.com/nestjs/nest/blob/master/packages/platform-express/adapters/express-adapter.ts#L210] and (fastify)[https://github.com/nestjs/nest/blob/master/packages/platform-fastify/adapters/fastify-adapter.ts#L455] adapters.

Reviewer 2: ```suggestion
#### Default body parser middleware

When using the `express` adapter, the NestJS app will register `json` and `urlencoded` from the package `body-parser` by default. This can be turned off by setting the `bodyParser` flag to `false` when creating the application. The options used when registering the middleware can be seen in the `registerParserMiddleware` function in the (express)[https://github.com/nestjs/nest/blob/master/packages/platform-express/adapters/express-adapter.ts#L210] and (fastify)[https://github.com/nestjs/nest/blob/master/packages/platform-fastify/adapters/fastify-adapter.ts#L455] adapters.
```

> #### Default body-praser middleware
> The nest application registers `json` and `urlencoded` from the package `body-parser` by default. This can be turned off by setting the `bodyParser` flag when creating the application. The options used when registering the middleware can be seen in the `registerParserMiddleware` function in the (express)[https://github.com/nestjs/nest/blob/master/packages/platform-express/adapters/express-adapter.ts#L210] and (fastify)[https://github.com/nestjs/nest/blob/master/packages/platform-fastify/adapters/fastify-adapter.ts#L455] adapters.

Reviewer 2: ```suggestion
#### Default body parser middleware

When using the `express` adapter, the NestJS app will register `json` and `urlencoded` from the package `body-parser` by default. This can be turned off by setting the `bodyParser` flag to `false` when creating the application.
```

Instead of adding links to the source, here (in the docs) we should mention those options.

Also, we should mention fastify adapter as well somehow, which doesn't use `body-parser` like the express adapter

Reviewer 2: tip: click in _details_ to see how the page will look like

![image](https://user-images.githubusercontent.com/13461315/184511107-647eb873-30c2-4876-9532-8abf588eb2d8.png)

```

> info **Hint** Accessing the DI container in a global middleware is not possible. You can use a [functional middleware](middleware#functional-middleware) instead when using `app.use()`. Alternatively, you can use a class middleware and consume it with `.forRoutes('*')` within the `AppModule` (or any other module).

#### Default body parser middleware

When using the `express` adapter, the NestJS app will register `json` and `urlencoded` from the package `body-parser` by default. 
- `urlencoded` is used with the `extended: true` flag and the optional `rawBody` flag if set in your `NestApplicationOptions`.
- `body-parser` is used with the optional `rawBody` flag if set in your `NestApplicationOptions`.

This can be turned off by setting the `bodyParser` flag to `false` when creating the application.

```typescript
const app = await NestFactory.create(AppModule, {bodyParser: false});
```

> Without doing that, you won't be able to apply custom bodyParser middleware to individual routes.

Reviewer 3: This isn't correct. AFAIR if you register a body-parser middleware by yourself - before initializing the application - the default one won't be registered, even without setting the `bodyParser` prop to false.
